validator/project_validator.go

```diff
@@ -584,6 +585,20 @@ func validateTaskNames(project *model.Project) ValidationErrors {
 	return errs
 }
 
+// validateModuleBranches checks to make sure that branches of modules have a set branch.
+func validateModuleBranches(project *model.Project) ValidationErrors {
+	errs := ValidationErrors{}
+	for _, module := range project.Modules {
+		if module.Branch == "" {
```

Comment:
maybe once we're validating modules we should also validate the name and repo fields?

Reply:
Sounds good to me. Would we only be checking that the name and repo also aren't empty strings? Or are there other requirements for a valid name and repo?

Reply:
For name it makes sense to me to validate 1) it's not the empty string (the empty string is a reserved module name), as you suggest. 2) there are no duplicates.
For repo at least that 1) it's not the empty string. Maybe also 2) that it conforms to one of the [Git URLs](https://git-scm.com/docs/git-clone#_git_urls), though the effort required may not be worth the benefit.

Reply:
It looks like we already have [a function for git URL validation](https://github.com/evergreen-ci/evergreen/blob/08d8cc896f586d8a23edf9bab940d6f2d3e70ffa/thirdparty/git.go#L268)